sdcm/nemesis.py

```diff
@@ -2696,15 +2696,84 @@ def modify_table_default_time_to_live(self):  # pylint: disable=invalid-name
             The value of this property is a number of seconds. If it is set, Cassandra applies a
             default TTL marker to each column in the table, set to this value. When the table TTL
             is exceeded, Cassandra tombstones the table.
+            This nemises selects random table, check if it has TimeWindowCompactionStrategy applied
+            and calculate possible default time to live and sets default values if table has no such strategy.
             default: default_time_to_live = 0
         """
         # Select table without columns with "counter" type for this nemesis - issue #1037:
         #    Modify_table nemesis chooses first non-system table, and modify default_time_to_live of it.
         #    But table with counters doesn't support this
 
         # max allowed TTL - 49 days (4300000) (to be compatible with default TWCS settings)
-        self._modify_table_property(name="default_time_to_live", val=random.randint(864000, 4300000),
-                                    filter_out_table_with_counter=True)
+
+        default_min_ttl = 864000  # 10 days in seconds
+        default_max_ttl = 4300000
+
+        ks_cfs = self.cluster.get_non_system_ks_cf_list(
+            db_node=self.target_node, filter_out_table_with_counter=True,
+            filter_out_mv=True)
+
+        keyspace_table = random.choice(ks_cfs) if ks_cfs else ks_cfs
+
+        def get_compaction_info_if_twcs_table(table):
+            keyspace, table = table.split('.')
+            query = f"SELECT compaction FROM system_schema.tables WHERE keyspace_name = '{keyspace}' AND table_name = '{table}';"
+
+            LOGGER.debug(f'Getting data from Scylla node: {self.target_node}, table: {table}')
+            with self.cluster.cql_connection_patient(self.target_node) as session:
+                result = session.execute(query)
+
+            for row in result:
+                if hasattr(row, 'compaction'):
+                    compaction_info_ = dict(row.compaction)
+                    if 'class' in compaction_info_ and compaction_info_['class'] == 'TimeWindowCompactionStrategy':
+                        return compaction_info_
+                    else:
+                        return None
+
+        def do_default_time_to_live_calculation(compaction_info_):
+            # Get compaction_window_size and unit
+            compaction_window_size = int(compaction_info_.get('compaction_window_size', '1'))
```

Comment:
If you are sure that it is TWCS, then all params in compaction_info_ must exist, using the Default value in this case is wrong. if there is an absent param it  must be an exception to indicate an SCT problem)

Reply:
Need to finalize such a decision